src/main/java/miu/cs425/controllers/VehicleController.java

```diff
@@ -27,4 +27,15 @@ public ResponseEntity<Car> addCar (@RequestBody Car car){
         Car newCar = carService.saveCar(car);
         return new ResponseEntity<>(newCar, HttpStatus.CREATED);
     }
+
+    @Operation(summary = "Retrieve all cars")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Return list all cars"),
+            @ApiResponse(responseCode = "204", description = "There is empty cars")
+    })
+    @GetMapping()
+    public ResponseEntity<List<Car>> getAllCars (){
+        List<Car> cars = carService.getAllCars();
+        return new ResponseEntity<>(cars, HttpStatus.OK);
```

Comment:
You need to check if cars is empty and return HttpStatusCode = 204 (NoContent) here.

Reply:
updated, please review again!